src/manager/PositionManager.ts

```diff
@@ -36,4 +43,45 @@ export class PositionManager {
     if (this.positions.has(key)) return this.positions.get(key)!;
     return new Position();
   }
+
+  collectPosition(
+    owner: string,
+    tickLower: number,
+    tickUpper: number,
+    amount0Requested: JSBI,
+    amount1Requested: JSBI
+  ): { amount0: JSBI; amount1: JSBI } {
+    assert(
+      JSBI.greaterThanOrEqual(amount0Requested, ZERO) &&
+        JSBI.greaterThanOrEqual(amount1Requested, ZERO),
+      "amounts requested should be positive"
+    );
+    const key = PositionManager.getKey(owner, tickLower, tickUpper);
+    if (this.positions.has(key)) {
+      const positionToCollect = this.positions.get(key)!;
+      const amount0 = JSBI.greaterThan(
+        amount0Requested,
+        positionToCollect.tokensOwed0
+      )
+        ? positionToCollect.tokensOwed0
+        : amount0Requested;
+      const amount1 = JSBI.greaterThan(
+        amount1Requested,
+        positionToCollect.tokensOwed1
+      )
+        ? positionToCollect.tokensOwed1
+        : amount1Requested;
+      assert(
+        JSBI.greaterThan(amount0, ZERO) && JSBI.greaterThan(amount1, ZERO),
+        "the amount to collect is invalid"
+      );
+      positionToCollect?.updateBurn(
+        JSBI.subtract(positionToCollect.tokensOwed0, amount0),
+        JSBI.subtract(positionToCollect.tokensOwed1, amount1)
+      );
+      if (positionToCollect.isEmpty()) this.clear(key);
```

Comment:
```
    if (JSBI.greaterThan(amount0, ZERO) || JSBI.greaterThan(amount1, ZERO)) {
      position.updateBurn(
        JSBI.subtract(position.tokensOwed0, amount0),
        JSBI.subtract(position.tokensOwed1, amount1)
      );
        if (positionToCollect.isEmpty()) this.clear(key);
    }
```

Reply:
I might prefer this since it's a simulator we are building and 0 amount is ok to collect in contract. And according to business logic it is allowed to collect either token0 or token1, or both of them in a request.